File: BackEnd/springboot.ecommerce/src/main/java/com/swayampp/ecommerce/dto/Purchase.java
package com.swayampp.ecommerce.dto;

import com.swayampp.ecommerce.entity.Address;
import com.swayampp.ecommerce.entity.Customer;
import com.swayampp.ecommerce.entity.Order;
import com.swayampp.ecommerce.entity.OrderItem;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Order order;
    private Address shippingAddr;
    private Address billingAddr;
    private Set<OrderItem> orderItems;
}

File: FrontEnd/angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    constructor(
        public customer: Customer,
        public order: Order,
        public shippingAddr: Address,
        public billingAddr: Address,
        public orderItems: OrderItem[]
    ) {}
}